team-telnyx/telnyx-webrtc-android
# [WEBRTC-690] Push notification and auto-login changes

[WebRTC-690 - Push notification issue investigation.](https://telnyx.atlassian.net/browse/WEBRTC-690)
---

This PR introduces a working auto-login once logged in before. If you have been logged in before, this will be flagged in shared preferences and login automatically. This means that once you receive a notification and launch the application on accepting a call, you will not have to manually log in again. 

There was an issue with this before in regards to auto login to dev or prod environments, with the preference not being remembered. The last used successfully logged into environment is now saved too, and an indicator is visible at the bottom right of the screen

This PR also includes some large TxSocket changes, simplifying our current solution. Tests have been adjusted to reflect this. 

## :older_man: :baby: Behaviors
### Before changes
Auto-login not working because of multiple environment use
TxSocket was over-engineered. 
No way to know if you are currently in Dev or Prod

### After changes
Auto-login now works regardless of the environment
TxSocket simplified
Dev/Prod indicator in small text bottom right

## ✋ Manual testing
1. Login in prod/dev
2. Close app. 
3. Relaunch.
4. You should auto-login with an indicator bottom right of the environment you last switched to and logged into
